lib/commodore.libjsonnet

```diff
@@ -55,13 +59,63 @@ local addNamespaceToHelmOutput(template_dir, namespace, exclude_objects=[]) =
   local input_file(elem) = template_dir + '/' + elem;
   local stem(elem) =
     local elems = std.split(elem, '.');
-    std.join('.', elems[:std.length(elems)-1]);
+    std.join('.', elems[:std.length(elems) - 1]);
   {
-    [stem(elem)]: patch_namespace(input_file(elem), namespace,
+    [stem(elem)]: patch_namespace(input_file(elem),
+                                  namespace,
                                   exclude_objects=exclude_objects)
     for elem in chart_files
   };
 
+/**
+* \brief Helper to inject proxy variables into a containers environment.
+*
+* HTTP proxy configuration is supposed to be done at `parameters.global` and
+* meant to be used by all components. This helper makes it easy to add those
+* values to a containers environment. No need to do any checks whether or not
+* they have to be added.
+*
+* This helper is suitable to be used with `env_:` from the Kubernetes Jsonnet
+* library. If the list form is used, combine it with `com.envList`.
+*
+* \return Dictionary. When configured, contains the http proxy environment
+*         variables in both upper and lower case form. Will be empty otherwise.
+*/
+local proxyVars = if std.objectHas(params, 'global') then {
+  [if std.objectHas(params.global, 'http_proxy') then 'HTTP_PROXY']: params.global.http_proxy,
+  [if std.objectHas(params.global, 'http_proxy') then 'http_proxy']: params.global.http_proxy,
+  [if std.objectHas(params.global, 'https_proxy') then 'HTTPS_PROXY']: params.global.https_proxy,
+  [if std.objectHas(params.global, 'https_proxy') then 'https_proxy']: params.global.https_proxy,
+  [if std.objectHas(params.global, 'no_proxy') then 'NO_PROXY']: params.global.no_proxy,
+  [if std.objectHas(params.global, 'no_proxy') then 'no_proxy']: params.global.no_proxy,
+} else {};
+
+/**
+* \brief Helper function to convert a dictionary into an environment list.
+*
+* Kubernetes containers require environment variables to be a list of objects.
+* In its simplest form, the object contains the keys `name` and `value`.
+*
+* This helper converts a dictionary into such a list where keys become `name`,
+* and values become `value`.
+*
+* \arg map Dictionary to be converted.
+*
+* \return List of dictionaries with `name`, `value` keys.
+*/
+local envList(map) = [
```

Comment:
I assume we need this for instances where we're not using `kube.libjsonnet`, as `kube.Container` supports setting `env_` and takes care of this conversion, see https://github.com/bitnami-labs/kube-libsonnet/blob/0e78b4332a893b7c1d03eefc84f9a2ef7d28b8ad/kube.libsonnet#L24-L40

Reply:
Exactly. E.g. when importing and manipulating an object from a static source or a Helm chart. I needed this in the ArgoCD component.